### ISSUE TITLE
Git dependencies - running build command does not work for local git repositories

### ISSUE THREAD

**Issue by theimowski** [fsprojects/Paket#1660]
### Description

When installing a local git repository as a NuGet source with a given build command, the build command is not invoked.

### Repro steps

```
git file:///c:\github\Argu paket_src_override build:"build.cmd NuGet", Packages: /bin/

nuget Argu
```

```paket install -v```

```
Paket version 3.0.0.0
found: c:\temp\paket.dependencies
Parsing c:\temp\paket.dependencies
99e32789711272310e53cf5642100f94e07980f1        refs/heads/paket_src_override
19a8144eb45d052b5c1a7402baa5a61e38f0a66d        refs/remotes/origin/paket_src_ov
erride
99e32789711272310e53cf5642100f94e07980f1
c:\github\Argu is already up-to-date.
Resolving packages for group Main:
  0 packages in resolution. 1 requirements left
  Trying to resolve Argu  (from c:\temp\paket.dependencies)
  - fetching versions for Argu
Paket failed with:
        The directory c:\github\Argu\bin doesn't exist.
Please check the NuGet source feed definition in your paket.dependencies file.
StackTrace:
     at Microsoft.FSharp.Core.PrintfModule.PrintFormatToStringThenFail@1360.Invo
ke(String message)
   at Paket.NuGetV2.getAllVersionsFromLocalPath@107-2.Invoke(String arg20) in C:
\github\Paket\src\Paket.Core\NuGetV2.fs:line 107
   at Paket.NuGetV2.getAllVersionsFromLocalPath@100.Invoke(Unit unitVar) in C:\g
ithub\Paket\src\Paket.Core\NuGetV2.fs:line 107
   at Microsoft.FSharp.Control.AsyncBuilderImpl.callA@851.Invoke(AsyncParams`1 a
rgs)
--- End of stack trace from previous location where exception was thrown ---
  (rest of stacktrace....) 
```

### Expected behavior

Build command (`build.cmd NuGet`) should be called before exploring the  `c:\github\Argu\bin` dir

### Actual behavior

Paket fails with `c:\github\Argu\bin is not found`

**Comment by theimowski** [comment]
Having a look at this now - I need it to work for #1636

**Comment by theimowski** [comment]
debugging showed that this has something to do with caching - when added `force` flag to install, the build command was run.
[here](https://github.com/fsprojects/Paket/blob/v3/src/Paket.Core/RemoteDownload.fs#L286-L289) `isInCorrectVersion` is true without `force`, and the build command is run in `downloadRemoteFiles` function

**Comment by theimowski** [comment]
fixed by #1669
